src/main/scala/se/lu/nateko/cp/meta/services/upload/DoiService.scala

```diff
@@ -0,0 +1,142 @@
+		new DoiClient(clientConf, http)
+	}
+
+	private val ccby4 = Rights("CC BY 4.0", Some("https://creativecommons.org/licenses/by/4.0"))
+	private val http = Http()
+
+	def saveDoi(meta: DoiMeta)(implicit envri: Envri): Future[Doi] = {
+		client.putMetadata(meta).map(_ => meta.doi)
+	}
+
+	def makeDataObjectDoi(uri: URI)(implicit envri: Envri): Future[DoiMeta] = {
+		fetchDataObject(uri).map{dobj =>
+			DoiMeta(
+				doi = client.doi(CoolDoi.makeRandom),
+				creators = dobj.references.authors.fold[Seq[Creator]](Seq())(_.map(a =>
+					Creator(PersonalName(a.firstName, a.lastName), Seq(), Seq()))
+				),
+				titles = dobj.references.title.map(t => Seq(Title(t, None, None))),
+				publisher = Some("ICOS ERIC -- Carbon Portal"),
+				publicationYear = Some(Year.now.getValue),
+				types = Some(ResourceType(None, Some(ResourceTypeGeneral.Dataset))),
+				subjects = Seq(),
+				contributors = dobj.specificInfo match {
+					case Left(l3) => l3.productionInfo.contributors.map(_ match {
+						case Person(_, firstName, lastName, _) => Contributor(PersonalName(firstName, lastName), Seq(), Seq(), None)
+						case Organization(_, name, _, _) => Contributor(GenericName(name), Seq(), Seq(), None)
+					})
+					case Right(_) => Seq()
+				},
+				dates = Seq(
+					Date(java.time.Instant.now.toString.take(10), DateType.Issued)
+				),
+				formats = Seq(),
+				version = Some(Version(1, 0)),
+				rightsList = Some(Seq(ccby4)),
+				descriptions = dobj.specificInfo match {
+					case Left(l3) => l3.description.map(d => Description(d, DescriptionType.Abstract, None)).toSeq
+					case Right(_) => Seq()
+				},
+				url = Some(uri.toString)
+			)
+		}
+	}
+
+	def makeDocObjectDoi(uri: URI)(implicit envri: Envri): Future[DoiMeta] = {
+		fetchDocObject(uri).map{dobj =>
+			DoiMeta(
+				doi = client.doi(CoolDoi.makeRandom),
+				titles = dobj.references.title.map(title => Seq(Title(title, None, None)))
+			)
+		}
+	}
+
+	def makeCollectionDoi(uri: URI)(implicit envri: Envri): Future[DoiMeta] = {
+		fetchCollectionObject(uri).map{coll =>
+			DoiMeta(
+				doi = client.doi(CoolDoi.makeRandom),
+				creators = Seq(Creator(GenericName(coll.creator.name), Seq(), Seq())),
+				titles = Some(Seq(Title(coll.title, None, None))),
+				publisher = Some("ICOS ERIC -- Carbon Portal"),
+				publicationYear = Some(Year.now.getValue),
+				types = Some(ResourceType(Some("ZIP archives"), Some(ResourceTypeGeneral.Collection))),
+				subjects = Seq(),
+				contributors = Seq(),
+				dates = Seq(
+					Date(java.time.Instant.now.toString.take(10), DateType.Issued)
+				),
+				formats = Seq(),
+				version = Some(Version(1, 0)),
+				rightsList = Some(Seq(ccby4)),
+				descriptions = coll.description.map(d => Description(d, DescriptionType.Abstract, None)).toSeq,
+				url = Some(uri.toString)
+			)
+		}
+	}
+
+	private def fetchDataObject(uri: java.net.URI): Future[DataObject] = http
```

Comment:
We should not be making HTTP calls for this, we are already inside meta's backend. Besides, these calls will go to the production server, so will not work for resources that have been created on a dev- or staging machine.